src/core/tsi/transport_security_interface.h

```diff
@@ -64,6 +64,26 @@ typedef enum {
   TSI_REQUEST_AND_REQUIRE_CLIENT_CERTIFICATE_AND_VERIFY,
 } tsi_client_certificate_request_type;
 
+typedef enum {
+  // TSI implementation provides a normal frame protector.  The caller
+  // should invoke tsi_handshaker_result_create_frame_protector() to
+  // generate the frame protector.
+  TSI_FRAME_PROTECTOR_NORMAL,
+  // TSI implementation provides a zero-copy frame protector.  The caller
+  // should invoke tsi_handshaker_result_create_zero_copy_grpc_protector()
+  // to generate the frame protector.
+  TSI_FRAME_PROTECTOR_ZERO_COPY,
+  // TSI implementation provides both normal and zero-copy frame protectors.
+  // The caller should invoke either
+  // tsi_handshaker_result_create_frame_protector() or
+  // tsi_handshaker_result_create_zero_copy_grpc_protector() to generate
+  // the frame protector.
+  TSI_FRAME_PROTECTOR_NORMAL_OR_ZERO_COPY,
+  // TSI implementation does not provide any frame protector.  This means
+  // that it is safe for the caller to send bytes unprotected on the wire.
+  TSI_FRAME_PROTECTOR_NONE,
+} tsi_frame_protector_type;
```

Comment:
If I understand correctly from our early discussion `frame_protector_type` API represents TSI implementor's intention on which frame protector a caller should use, instead of merely reflecting the fact that which frame protectors get implemented. So for instance, a specific TSI implementation can let `frame_protector_type` API return `TSI_FRAME_PROTECTOR_NORMAL` even if both frame protectors are implemented, just to tell callers that they should only use the regular frame protector.  Similarly, if TSI implementation returns `TSI_FRAME_PROTECTOR_NORMAL_OR_ZERO_COPY`, then it tells callers both frame protectors are available and they can use whatever they want (for gRPC, it uses zero-copy frame protector). Is my understanding correct?